orttraining/orttraining/python/training/_checkpoint_storage.py

```diff
@@ -79,3 +80,17 @@ def load(path, key=None):
         _dfs_load(f, load_obj)
 
     return load_obj
+
+def to_serialized_hex(user_dict):
+    """Serialize the user_dict and convert the serialized bytes to a hex string and return"""
+
+    return pickle.dumps(user_dict).hex()
+
+def from_serialized_hex(serialized_hex):
+    """Convert serialized_hex to bytes and deserialize it and return"""
+
+    try:
+        serialized_hex = serialized_hex.decode()
+    except AttributeError:
+        pass
```

Comment:
why it is ok to ignore AttributeError? Add at least a comment on the reasoning

Reply:
```serialized_hex``` must be a string for ```bytes.fromhex()``` to succeed. But the input can also be a byte string. In which case, I need to first convert to a regular string.
If the input is a regular string, it does not have the attribute ```decode()```. So what this is doing is that try to decode the byte string. If an AttributeError is encountered, then the string must be a regular string, so do nothing.